Jenkinsfile

```diff
@@ -0,0 +1,114 @@
+        	OUTPUT_FOLDER = "diagram"
+    	}
+	stages{
+		// This stage checks that upstream project 'GenerateGraphDatabaseAndAnalysisCore' was run successfully.
+		stage('Check Graph DB & Analysis Core build succeeded'){
+			steps{
+				script{
+                    			utils.checkUpstreamBuildsSucceeded("GenerateGraphDatabaseAndAnalysisCore")
+				}
+			}
+		}
+		// This stage builds the jar file using maven.
+		stage('Setup: Build jar file'){
+			steps{
+				script{
+					sh "mvn clean package"
+				}
+			}
+		}
+		// Execute the jar file, producing the diagram JSON files.
+		stage('Main: Run Diagram-Converter'){
+			steps{
+				script{
+				    def releaseVersion = utils.getReleaseVersion()
+					sh "mkdir -p ${env.OUTPUT_FOLDER}"
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'mysqlPass', usernameVariable: 'mysqlUser')]){
+						withCredentials([usernamePassword(credentialsId: 'neo4jUsernamePassword', passwordVariable: 'neo4jPass', usernameVariable: 'neo4jUser')]){
+				 			sh "java -jar target/diagram-converter-jar-with-dependencies.jar --graph_user $neo4jUser --graph_password $neo4jPass --rel_user $mysqlUser --rel_password $mysqlPass --rel_database ${env.REACTOME_DB} --output ./${env.OUTPUT_FOLDER}"
+					        	// Create archive that will be stored on S3.
+							sh "tar -zcf diagrams-v${releaseVersion}.tgz ${env.OUTPUT_FOLDER}/"
+							// Restart tomcat7 and neo4j services after updates were made to graph db.
+					        	sh "sudo service tomcat7 stop"
+							sh "sudo service neo4j stop"
+							sh "sudo service neo4j start"
+							sh "sudo service tomcat7 start"
+						}
+					}
+				}
+			}
+		}
+		// There are generally over 30k JSON diagram files produced in a typical release.
+		// This stage gets the file counts between the current and previous release, which allows for quick review.
+		stage('Post: Compare previous release file number') {
+		    steps{
+		        script{
+				def releaseVersion = utils.getReleaseVersion()
+				def previousReleaseVersion = utils.getPreviousReleaseVersion()
+				def previousDiagramsArchive = "diagrams-v${previousReleaseVersion}.tgz"
+				sh "mkdir -p ${previousReleaseVersion}"
+				// Download previous diagram-converter output files and extract them.
+				sh "aws s3 --no-progress cp s3://reactome/private/releases/${previousReleaseVersion}/diagram_converter/${previousDiagramsArchive} ${previousReleaseVersion}/"
+				dir("${previousReleaseVersion}"){
+					sh "tar -xf ${previousDiagramsArchive}"
+				}
+				// Output number of JSON diagram files between releases.
+				def currentDiagramsFileCount = findFiles(glob: "${env.OUTPUT_FOLDER}/*").size()
+				def previousDiagramsFileCount = findFiles(glob: "${previousReleaseVersion}/${env.OUTPUT_FOLDER}/*").size()
+				echo("Total diagram files for v${releaseVersion}: ${currentDiagramsFileCount}")
+				echo("Total diagram files for v${previousReleaseVersion}: ${previousDiagramsFileCount}")
```

Comment:
Can/should there be an automatic check to make sure the file counts are equal or if there's a change it's expected?

Reply:
I think thats a good idea in the future, but really this is just supposed to provide a shortcut for evaluating that the step ran correctly - normally for this step you just confirm the file counts are _around_ the same (if not slightly increased) between releases. I know for the Perl version of things that we had a 10% threshold for things like orthopairs, but we also had issues with automating that check.

Reply:
Okay, just a thought.